### DISCUSSION TITLE
Softdelete

### DISCUSSION THREAD

**Discussion by lloricode** [rappasoft/laravel-livewire-tables#244] [Q&A] [answered]
Hi how can we able to fetch trashed data?
and also what is better approach? 

I'm new in liveware but able to understand, since its purely back-end, but no clue how can dynamically get trashed data

thank you

**Comment by bdelamatre** [comment, marked as answer]
This isn't a feature that's specifically in scope for this project, but I assume an approach like this:

1. Add a filter to include soft deleted records
2. Adjust your query to use the soft deleted filter.

**Reply by lloricode** [reply]
> This isn't a feature that's specifically in scope for this project, but I assume an approach like this:
> 
> 1. Add a filter to include soft deleted records
> 2. Adjust your query to use the soft deleted filter.

thank you, will try this later

**Reply by rappasoft** [reply]
> This isn't a feature that's specifically in scope for this project, but I assume an approach like this:
> 
> 1. Add a filter to include soft deleted records
> 2. Adjust your query to use the soft deleted filter.

That's how I would do it.

**Reply by lloricode** [reply]
@rappasoft @bdelamatre can you add sample in wiki? since this is the common filter will be use
im struggle how to make filter with softdelete,
![Screenshot from 2021-04-30 08-31-57](https://user-images.githubusercontent.com/8251344/116634421-8cf13400-a98e-11eb-9c89-03015fc0fe3e.png)
also why there a dropdown? im still confusing

thank you

**Reply by lloricode** [reply]
already done, its just my IDE did not suggesting 
```php
$query->onlyTrashed();

$query->withTrashed();
```

thank you

**Comment by rubensrocha** [comment]
Example: 

```
public function filters(): array{
        return [
            SelectFilter::make('Deleted')->options(['' => 'No', 'yes' => 'Yes'])
            ->filter(function(Builder $builder, string $value) {
                if ($value === 'yes') {
                    $builder->onlyTrashed();
                }
            })
        ];
    }
```
